pmm/summary.go

```diff
@@ -0,0 +1,263 @@
+/*
+	Copyright (c) 2016, Percona LLC and/or its affiliates. All rights reserved.
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>
+*/
+
+package pmm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+	"regexp"
+	"strings"
+	"io"
+	"archive/zip"
+	"path/filepath"
+)
+
+// Collector parameters and description.
+type Collector struct {
+	CollectorDescription	string
+	ExecCommand		[]string
+	OutputFileName		string
+}
+
+// CollectData runs a command and collects output into a file.
+func (c Collector) CollectData() {
+	fmt.Printf("%s ... ",c.CollectorDescription)
+	dstfNetworks, err := os.OpenFile(c.OutputFileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0777)
+	if err != nil {
+		fmt.Printf("Skipped. Failed to create file %s with %s\n", c.OutputFileName, err)
+	} else {
+		defer dstfNetworks.Close()
+		cmd, err := exec.Command(c.ExecCommand[0], c.ExecCommand[1:]...).Output()
+		if err != nil {
+			fmt.Printf("Skipped. Error collecting data by %s: %s\n",c.ExecCommand,err)
+		} else {
+			if _, err := dstfNetworks.Write(cmd); err != nil {
+				fmt.Printf("Skipped. Failed to copy %s output into file %s\n",c.ExecCommand[0],c.OutputFileName)
+			} else {
+				fmt.Println("Done.")
+			}
+		}
+	}
+}
+
+// CheckInitType detects which system manager is running on Linux System.
+func CheckInitType() string {
+	var initType string
+	cmdSystemType, err := exec.Command("stat", "/proc/1/exe").Output()
+	if err != nil {
+		fmt.Println("Error exec command stat /proc/1/exe:", err)
+	} else {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow There and in a few other places.

Reply:
Decreased indentation